tensorflow/core/lib/strings/proto_serialization.cc

```diff
@@ -72,7 +72,7 @@ bool SerializeToBufferDeterministic(const protobuf::MessageLite& msg,
   protobuf::io::CodedOutputStream output_stream(&array_stream);
   output_stream.SetSerializationDeterministic(true);
   msg.SerializeWithCachedSizes(&output_stream);
-  return !output_stream.HadError() && size == output_stream.ByteCount();
+  return !output_stream.HadError() && size == (size_t)(output_stream.ByteCount());
```

Comment:
Let's use `static_cast` instead of old C-style casts:

```suggestion
  return !output_stream.HadError() && size == static_cast<size_t>(output_stream.ByteCount());
```